keras/src/ops/nn.py

```diff
@@ -2651,3 +2651,68 @@ def dot_product_attention(
         is_causal=is_causal,
         flash_attention=flash_attention,
     )
+
+
+class RMSNorm(Operation):
+    def __init__(self, scale, axis=-1, epsilon=None):
+        super().__init__()
+        self.axis = axis
+        self.scale = scale
+        self.epsilon = epsilon
+
+    def compute_output_spec(self, x):
+        return KerasTensor(shape=x.shape)
+
+    def call(self, x):
+        return _normalize(
+            x, scale=self.scale, axis=self.axis, epsilon=self.epsilon
+        )
+
+
+@keras_export(
+    [
+        "keras.ops.rms_norm",
+        "keras.ops.nn.rms_norm",
+    ]
+)
+def rms_norm(x, scale=1, axis=-1, epsilon=None):
```

Comment:
You can name it `rms_normalization` (no abbreviations is better)

If it's a stateless op, it doesn't need to be a layer. Doing `x = rms_normalization(x)` will work in both a call context and during functional model construction (strictly equivalent to `x = RMSNormalization()(x)`)

The only motivation for having the layer would be UX -- does it provide a UX benefit?

Reply:
But is the state trainable -- is it a weight?

Reply:
Strictly speaking - yes, in the same way gamma/beta are in LayerNormalization are:

```
>>> import keras
>>> layer = keras.layers.RMSNormalization(input_dim=10)
>>> print(layer.scale.shape)
(10,)
>>> print(layer.scale.trainable)
True
```

And:

```
>>> layer = keras.layers.LayerNormalization(axis=[1, 2, 3])
>>> layer.build([5, 20, 30, 40])
>>> print(layer.beta.shape)
(20, 30, 40)
>>> print(layer.beta.trainable)
True
```

Or did I miss the question? 😓